lib/stackify/unix_socket_sender.rb

```diff
@@ -0,0 +1,69 @@
+require 'net_http_unix'
+require 'ostruct'
+
+#
+# This class will handle the sending of protobuf message to unix domain socket
+#
+module Stackify
+  class UnixSocketSender < Worker
+
+    # send_logs() Function to put the msg in the Worker
+    def send_logs msgs, attempts = 3
+      worker = Stackify::LogsSenderWorker.new('UnixSocketSender worker')
+      task = send_logs_task attempts, msgs
+      worker.async_perform ScheduleDelay.new, task
+    end
+
+    private
+
+    def properties
+      {
+        success_condition: lambda { |result| result.try(:status) == 200 },
+        limit: 1
+      }.dup
+    end
+
+    def send_logs_task attempts = nil, msgs
+      properties[:attempts] = attempts if attempts
+      Stackify::ScheduleTask.new properties do
+        send_request msgs
+      end
+    end
+
+    # send_request() This function will send http request via unix domain socket
+    # @msgs {Object} Protobuf message
+    # return {Object} Return an object {status, message}
+    def send_request msgs
+      begin
+        details = Stackify::EnvDetails.instance.set_rails_info
+        log_group = Stackify::LogGroup.new
```

Comment:
suggest refactoring to have the log_group creation outside of the send_requests function.